src/streaming/controllers/BufferController.js

```diff
@@ -485,21 +485,17 @@ function BufferController(config) {
     function pruneBuffer() {
         if (type === 'fragmentedText') return;
 
-        var bufferToPrune = 0;
+        var start = buffer.buffered.length ? buffer.buffered.start(0) : 0;
```

Comment:
I checked and initially I agreed, but I believe the key part here is ```if (bufferToPrune > 0)``` - this would always be > 0 once we hit the prune threshold resulting in pruning potentially far more often than necessary.

I also think that if we consider reintroducing minimum buffer to prune that we'd then also need a similar check so is worthwhile overall.